src/EditorFeatures/CSharpTest2/Recommendations/AssemblyKeywordRecommenderTests.cs

```diff
@@ -115,6 +116,97 @@ public async Task TestInOuterAttribute()
             await VerifyKeywordAsync(
 @"[$$");
         }
+        
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        [WorkItem(362, "https://github.com/dotnet/roslyn/issues/362")]
+        public async Task TestBeforeAttributeNamespace()
+        {
+            await VerifyKeywordAsync(
+@"[$$
+namespace Goo {}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        [WorkItem(362, "https://github.com/dotnet/roslyn/issues/362")]
+        public async Task TestNotBeforeAttributeNamespaceWithNoOpenBracket()
+        {
+            await VerifyAbsenceAsync(
+@"$$
+namespace Goo {}");
+        }
+
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        [WorkItem(362, "https://github.com/dotnet/roslyn/issues/362")]
+        public async Task TestNotBeforeAttributeNamespaceAndAfterUsingWithNoOpenBracket()
+        {
+            await VerifyAbsenceAsync(
+@"
+using System;
+
+$$
+namespace Goo {}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        [WorkItem(362, "https://github.com/dotnet/roslyn/issues/362")]
+        public async Task TestBeforeAttributeNamespaceAndAfterUsingWithNoOpenBracket()
+        {
+            await VerifyKeywordAsync(
+@"
+using System;
+
+[$$
+namespace Goo {}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        [WorkItem(362, "https://github.com/dotnet/roslyn/issues/362")]
+        public async Task TestBeforeAssemblyAttributeWithOpenBracket()
+        {
+            await VerifyKeywordAsync(
+@"
+[$$
+[assembly: Whatever]
+namespace Goo {}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        [WorkItem(362, "https://github.com/dotnet/roslyn/issues/362")]
+        public async Task TestNotBeforeAssemblyAttributeWithoutOpenBracket()
+        {
+            await VerifyAbsenceAsync(
+@"
+$$
+[assembly: Whatever]
+namespace Goo {}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        [WorkItem(362, "https://github.com/dotnet/roslyn/issues/362")]
+        public async Task TestNotBeforeAssemblyAttributeAndAfterUsingWithoutOpenBracket()
+        {
+            await VerifyAbsenceAsync(
+@"
+using System;
+
+$$
+[assembly: Whatever]
+namespace Goo {}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        [WorkItem(362, "https://github.com/dotnet/roslyn/issues/362")]
+        public async Task TestBeforeAssemblyAttributeAndAfterUsingWithoutOpenBracket()
+        {
+            await VerifyKeywordAsync(
+@"
+using System;
+
+[$$
+[assembly: Whatever]
+namespace Goo {}");
```

Comment:
another test needs to be that this does *not* show up before a *nested* class

Reply:
like this?
`[$$
class A {}
class B {}`

Reply:
more like:

```c#
class A
{
    [$$
}
```

```c#
class A
{
    [$$
    class B
    {
    }
}
```

Reply:
Should be possible use `assembly` before class/interface/struct/enum? Would you like I create more test with these case?

``` csharp
[$$
class A {}
```
``` csharp
[$$
interface A {}
```
``` csharp
[$$
struct A {}
```
``` csharp
[$$
enum A {}
```

Reply:
> Should be possible use assembly before class/interface/struct/enum?

Yes.  It would be legal for a user to write:

```c#
[assembly: Whatever]

class C
{
}
```

> Would you like I create more test with these case?

The more hte merrier :)